enhancements/network/kubernetes-nmstate-operator-to-cvo.md

```diff
@@ -0,0 +1,284 @@
+---
+title: add-kubernetes-nmstate-operator-to-cvo
+authors:
+  - "@bcrochet"
+  - "@yboaron"
+  - "@hardys"
+reviewers:
+  - "@cgwalters"
+  - "@derekwaynecarr"
+  - "@russellb"
+approvers:
+  - TBD
+creation-date: 2021-03-16
+last-updated: 2021-04-23
+status: provisional
+see-also:
+  - "/enhancements/machine-config/mco-network-configuration.md"
+---
+
+# Add Kubernetes NMState Operator to release payload
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Enable Kubernetes NMstate by default for selected platforms.
+
+Network interface configuration is a frequent requirement for some platforms, particularly baremetal.
+In recent releases [work was completed]("/enhancements/machine-config/mco-network-configuration.md")
+to enable Kubernetes NMstate (optionally via OLM) so that declarative configuration
+of secondary network interfaces is possible.
+
+To improve the user-experience it is desirable to enable the NMState API by default on selected platforms
+where such configuration is common.
+
+## Motivation
+
+In previous discussion it was noted that we may [not want NMState enabled on all platforms](https://github.com/openshift/enhancements/pull/161#discussion_r433303754) and also that [all tech-preview features must require opt-in](https://github.com/openshift/enhancements/pull/161#discussion_r433315534).
+
+Now, the support status of NMState is moving from tech-preview to fully supported, the main consideration is how to conditionally enable this only on platforms where it's likely to be needed.
+
+If we can enable Kubernetes NMstate as part of the release payload, we can improve the user experience on platforms where it is required - for example allow for NodeNetworkConfigurationPolicy to be provided via manifests at install-time via openshift-installer.
```

Comment:
In #753 we are addressing a similar need with the performance-addon-operator. In that case, we're exploring the option of running the operator's `render` command outside of the installer between the `create manifests` and `create cluster` phases. Could a similar approach be used with kubernetes-nmstate?

Reply:
Yes I think that could work, if the needed containers are in the release payload then even if we require an "advanced" interface at the installer phase it's an improvement on the current process where everything has to happen day-2 via OLM.  @yboaron would you agree?

Reply:
Yep, that sounds like a good direction.